navigator_api/api/main.py

```diff
@@ -93,79 +96,55 @@ def workflow_state(dataset_id):
     workflow = get_or_create_workflow(dataset['id'], current_user.id, name=dataset['title'])
     if not workflow:
         return error.error_response(500, f"Couldn't get decision engine details for dataset {dataset_id}")
-
+    engine_decision = engine_client.get_decision(ckan_cli, dataset_id, skip_actions=workflow.skipped_tasks)
     return jsonify({
         "id": f"{workflow.id}",
-        "milestones": [
-            {
-                "id": "xxx",
-                "title": "Naomi Data Prep",
-                "completed": True,
-                "progress": 100
-            },
-            {
-                "id": "yyy",
-                "title": "Shiny 90 Data Prep",
-                "completed": False,
-                "progress": 50
-            },
-            {
-                "id": "zzz",
-                "title": "Update Spectrum",
-                "completed": False,
-                "progress": 0
-            }
-        ],
-        "taskBreadcrumps": [
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa",
-            "aaabbbbcc",
-            "dddeeefff",
-            "asdfasdfa"
-        ],
+        "progress": engine_decision["progress"]["progress"],
+        "milestones": engine_decision["progress"]["milestones"],
+        "milestoneListFullyResolved": engine_decision["progress"]["milestoneListFullyResolved"],
+        "taskBreadcrumps": engine_decision["actions"],
         "currentTask": {
-            "id": "asdfasdfa",
-            "skipped": False,
-            "details": {
-                "milestoneId": "zzz",
-                "title": "Populate ART template",
-                "displayHtml": "<p><strong>Lorem Ipsum</strong> is simply dummy <br /> "
-                               "text of the printing and typesetting industry.</p>",
-                "skippable": True,
-                "helpUrls": [
-                    {"label": "Naomi help docs", "url": "http://example"},
-                    {"label": "Spectrum documentation", "url": "http://example"}
-                ]
-            }
+            "id": engine_decision["decision"]["id"],
+            "skipped": is_task_skipped(dataset_id, engine_decision["decision"]["id"]),
```

Comment:
I've been thinking... We need to add `removeSkips: [<node_id>]` to the engine response (or some similar solution) if the engine is recommending that you should remove some skipped steps in order to proceed. 

So if the engine recommends removing any skips, you should update the user state accordingly in the api layer.   This is essential for the skip logic to work.

Reply:
Yeah, that sounds like a separate, longer discussion.